### PR TITLE
Normalize scores

### LINKED ISSUE #102: Normalize metric scores before applying weights in ETL
Currently we are deriving the Impact Index without considering the maximum values for each project. I think what needs to be done is we need to find the max value out of all the projects (for each metric) and adjust the scores accordingly. Then we should multiply by the weights and that result for each metric to the impact_index for that particular day. This should be similar to how we did it in Impact Calculator.

### PR DESCRIPTION
**Pull request by swellander**
## Description
Updates our ETL process to normalize projects' scores while calculating `impact_index`. Closes #102

### CODE REVIEW

File: packages/nextjs/pages/api/etl/index.ts [hunk 1]
```diff
@@ -267,21 +271,29 @@ const getMovement = (current: number, comparison: number) => {
...
+  for (const metric of metrics) {
     const multiplier = indPct ? weights[metric] / weightSum : weights[metric];
-    // Normalize the weights to their percentage of the total
-    const projectMetric = project[metric as unknown as keyof Metrics];
-    if (projectMetric) {
-      impact_index += projectMetric * multiplier;
+
+    // Normalize score
+    const max = maxScoreByMetric[metric] || 0;
+    const normalizer = max !== 0 ? 100 / max : 0;
```
Comment (swellander) [inline packages/nextjs/pages/api/etl/index.ts:290]: This was taken from our [Impact Calculator implementation](https://github.com/BuidlGuidl/impact-calculator/blob/e6cd7f474abb0ad67df582c59748343ea0ef72eb/packages/nextjs/pages/api/impact/index.ts#L180)

File: packages/nextjs/pages/api/etl/index.ts [hunk 2]
```diff
@@ -267,21 +271,29 @@ const getMovement = (current: number, comparison: number) => {
...
   let weightSum = 0;
-  for (const metric of metricKeys) {
+  for (const metric of metrics) {
     weightSum += weights[metric];
   }
   // If the weights don't add up to 100 then we expect they are each a 0-100 percentage and need to be adjusted
   const indPct = Math.round(weightSum) !== 100;
-  for (const metric of metricKeys) {
+  for (const metric of metrics) {
     const multiplier = indPct ? weights[metric] / weightSum : weights[metric];
```
Comment (escottalexander) [inline packages/nextjs/pages/api/etl/index.ts]: Remove the condition logic here. This is left over from before we knew the format of the weights.
You can remove line 286 (+ the comment above it) and make this line look like this:
```javascript
const multiplier = weights[metric] / weightSum;
```

**Review by escottalexander — CHANGES_REQUESTED** [review]
I checked this over and gave it careful thought and I think you have got it. I did leave a comment on removing some old logic that I added awhile that has the potential to mess things up.

File: packages/nextjs/pages/api/etl/index.ts (hunk 2 above)
Comment (swellander) [reply packages/nextjs/pages/api/etl/index.ts]: Makes sense, thanks

**Review by escottalexander — APPROVED** [review]
Perfect.

**Pull request merged**